### PR TITLE
Generate Exceptional Exceptions

### PR DESCRIPTION
**Pull request by anvacaru**
Fixes: https://github.com/runtimeverification/firefly/issues/568

### CODE REVIEW

File: tests/web3/invalidOp0.expected.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        "jsonrpc": "2.0",
+        "id": 3,
+        "error": {
+            "message": "VM Exception while processing transaction: EVMC_UNDEFINED_INSTRUCTION",
+            "code": -32000,
+            "data": {
+                "0x5daefe6da6e340448ea5c8ff3926de2aa48e4423476a438f1533a18b4347c25f": {
+                    "error": "EVMC_UNDEFINED_INSTRUCTION",
+                    "program_counter": 1,
```
Comment (anvacaru) [inline tests/web3/invalidOp0.expected.json]: There might be an issue with the returned `program_counter`.

File: web3.md [hunk 2]
```diff
@@ -630,13 +630,15 @@ eth_sendTransaction
            ...
          </account>
 
-    syntax JSON ::= #generateException( String, Int, Bytes ) [function]
- // -------------------------------------------------------------------
-    rule #generateException(TXHASH, PCOUNT, RD) => { "message": "VM Exception while processing transaction: revert",
+    rule <k> #loadNonce _ _ => 0 ... </k> [owise]
```
Comment (ehildenb) [inline web3.md]: Why is this here? Why does it put an `Int` on the `<k>` cell when the normal rule for `#loadNonce` doesn't put anything on the cell?

File: tests/web3/undefinedOpcode.expected.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        "jsonrpc": "2.0",
+        "id": 2,
+        "result": true
+    },
+    {
+        "jsonrpc": "2.0",
+        "id": 3,
+        "error": {
+            "message": "VM Exception while processing transaction: EVMC_UNDEFINED_INSTRUCTION",
```
Comment (ehildenb) [inline tests/web3/undefinedOpcode.expected.json]: I feel like this should be `invalid opcode` instead of `EVMC_UNDEFINED_INSTRUCTION` here.

**Comment by ehildenb** [comment]
Is there any reason for `eth_sendTransaction_final` to not use the same `#generateException` function?

File: web3.md (hunk 2 above)
Comment (anvacaru) [reply web3.md]: This was a typo.

File: tests/web3/undefinedOpcode.expected.json (hunk 3 above)
Comment (ehildenb) [reply tests/web3/undefinedOpcode.expected.json]: NVM, this is fine.

**Comment by ehildenb** [comment]
Add a test for `EVMC_INVALID_OPCODE` because that's also changing in this PR.
